File: LeetCode/Easy/Problems 851-900/Probelm 876/p876Sol1.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def middleNode(self, head: Optional[ListNode]) -> Optional[ListNode]:
        currentNode = head
        length = 0
        
        while currentNode is not None:
            currentNode = currentNode.next
            length += 1
            
        solHead = head
        for i in range(length//2):
            solHead = solHead.next
            
        return solHead

File: LeetCode/Easy/Problems 851-900/Probelm 876/p876Sol1.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */


struct ListNode* middleNode(struct ListNode* head){
    struct ListNode* currentNode = head;
    int length = 0;
    
    while (currentNode != NULL) {
        currentNode = currentNode->next;
        length++;
    }
    
    struct ListNode* solHead = head;
    for (int i = 0; i < length/2; i++) {
        solHead = solHead->next;
    }
    
    return solHead;
}